internal/mode/static/usage/secret.go

```diff
@@ -0,0 +1,60 @@
+package usage
+
+import (
+	"sync"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 . SecretStore
+
+// SecretStore should store the usage Secret that contains the credentials for NGINX Plus usage reporting.
+type SecretStore interface {
+	// Set stores the updated Secret.
+	Set(*v1.Secret)
+	// Delete nullifies the Secret value.
+	Delete()
+	// GetCredentials returns the base64 encoded username and password from the Secret.
+	GetCredentials() ([]byte, []byte)
+}
+
+// SecretStoreImpl implements the SecretStore interface.
+type SecretStoreImpl struct {
```

Comment:
I know we use `impl` all over our code, but it's really an anti-pattern, especially since interfaces are supposed to be defined on the client-side. 

I suggest naming this UsageCredentials or UsageSecret.

Reply:
I can do Secret, since the package name is `usage`, it'll stutter.